test/common/pixel-tests

```diff
@@ -1,25 +1,35 @@
 #!/bin/bash
 
-set -eu
-
-SUBDIR=test/reference
-REPO=pixel-test-reference
-
-GITHUB_BASE="${GITHUB_BASE:-cockpit-project/cockpit}"
-GITHUB_REPOSITORY="${GITHUB_BASE%/*}/${REPO}"
-CLONE_REMOTE="https://github.com/${GITHUB_REPOSITORY}"
-PUSH_REMOTE="git@github.com:${GITHUB_REPOSITORY}"
+GITHUB_REPO='pixel-test-reference'
+SUBDIR='test/reference'
 
-message() {
-    [ "${V-}" != 0 ] || printf "  %-8s %s\n" "$1" "$2"
-}
+set -eu
+cd "$(realpath -m "$0"/../../..)"
+. pkg/lib/git-utils.sh
 
 cmd_init() {
     git submodule add -b empty "$CLONE_REMOTE" "$SUBDIR"
 }
 
+cmd_checkout() {
+    local sha="${1-$(get_index_gitlink node_modules)}"
+
+    fetch_sha_to_cache "${sha}"
+    cmd_remove
+    clone_from_cache "${sha}"
+}
+
 cmd_pull() {
-    git submodule update --init -- "$SUBDIR"
+    # Our main goal is to ensure that the test/reference from
+    # the index is the one that we actually have.
+    local sha="$(get_index_gitlink test/reference)"
+    if [ ! -e test/reference/.git ]; then
+        # nothing there yet...
+        cmd_checkout "${sha}"
+    elif [ "$(git -C test/reference rev-parse HEAD)" != "${sha}" ]; then
+        # wrong thing there...
+        cmd_checkout "${sha}"
```

Comment:
Looks a bit odd to have two branches with exactly the same code. This could just `||` the condition.